src/core/processing/qgsprocessingparameters.cpp

```diff
@@ -67,6 +67,28 @@ bool QgsProcessingFeatureSourceDefinition::loadVariant( const QVariantMap &map )
 }
 
 
+//
+// QgsProcessingRasterLayerDefinition
+//
+
+QVariant QgsProcessingRasterLayerDefinition::toVariant() const
```

Comment:
please add tests for all the new QgsProcessingRasterLayerDefinition related logic